Adjusts XYZ to Mol and XYZ File to Mol modules to assume angstrom and convert to bohr

jlheflin opened the pull request:
**Is this pull request associated with an issue(s)?**
No

**Description**
Converts angstrom-based XYZ files to bohr coordinates for Chemist conventions.

**TODOs**
None, r2g.

jlheflin committed:
clang-format changes

jwaldrop107 reviewed (changes requested):
This is on the right track, but I think we'll want to go about it differently. Instead of changing the default behavior, I would add an input to the module called `"units"` that defaults to `"bohr"`. Then have the module check the value of that input to determine if the coordinate values need to be converted (and throw on non-permissible values).

ryanmrichard commented on src/cxx/nux/xyz_to_mol.cpp:
```diff
@@ -20,6 +20,12 @@
 
+double ang2bohr(double ang_value) {
+    // Converts a anstrom-based value to a bohr-based value
+    double bohr_value = ang_value * 1.8897259886;
```
This is what's called a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)). They're considered an anti-pattern. What I would suggest is to add a module specific input to `XYZToMolecule` called `to bohr conversion factor` (or something like that) which is the conversion to apply to take the input coordinates and convert them to bohr. Like @jwaldrop107 suggested, I would make it so the default behavior remains the same, i.e.,. set the conversion factor to 1.0.

jlheflin committed:
clang format changes

jlheflin committed:
added testing for the x and y components instead of just the z component of the xyz to molecule

jlheflin commented:
I added in the unit scaling factor, I also "stress tested" the XYZ File to Molecule module with both a converted protein pdb from RCSB (4EMU, 2977 atoms) as well as an OpenMM generated solvent box, converted to an xyz (98880 atoms). The 4EMU took 290 ms, and the solvated box took 56952 ms.
